addons/mass_mailing/models/mail_blacklist.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class MailBlackList(models.Model):
+    """ Model of blacklisted email addresses to stop sending emails."""
+    _inherit = ['mail.blacklist']
+
+    opt_out_reason_id = fields.Many2one(
+        'mailing.subscription.optout', string='Reason',
+        ondelete='cascade',
```

Comment:
Not sure.
We clearly do not want to remove people from the blacklist if someone deletes the opt-out reason.
Either
```suggestion
        ondelete='set null',
```
Or even
```suggestion
        ondelete='restrict',
```

Reply:
This should be a restrict, thanks :) we don't want to loose information about this, as it can be used notably if we have gdpr complaints or stuff like that.

Reply:
We may need an `active` field on `mailing.subscription.optout`, then.